File: Application/App/src/app/models/matchDetails/perks-dto.ts
import { PerkStatsDto } from "./perk-stats-dto";
import { PerkStyleDto } from "./perk-style-dto";

export class PerksDto {
    statPerks : PerkStatsDto = new PerkStatsDto({});
    styles : PerkStyleDto[] = [];

    public constructor(data: Partial<PerksDto>){
        Object.assign(this,data);
    }
}

File: Backend/Backend/Models/Matches/PerksDto.cs
namespace Backend.Models.Matches
{
    public class PerksDto
    {
        public PerksDto(PerkStatsDto statPerks, List<PerkStyleDto> styles)
        {
            this.StatPerks = statPerks;
            this.Styles = styles;
        }

        public PerkStatsDto StatPerks { get; set; }
        public List<PerkStyleDto> Styles { get; set; }
    }
}
